In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("/Users/alexmahadevan/Code/raise_a_family/INITIAL_RAISE_A_FAMILY_DF.csv")

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3221 entries, 0 to 3220
Data columns (total 25 columns):
Unnamed: 0                                   3142 non-null float64
SUMLEV                                       3142 non-null float64
REGION                                       3142 non-null float64
DIVISION                                     3142 non-null float64
STATE                                        3142 non-null float64
COUNTY                                       3142 non-null float64
STNAME                                       3142 non-null object
CTY                                          3142 non-null object
Unemployment_rate_2016                       3142 non-null float64
Med_HH_Income_Percent_of_State_Total_2015    3142 non-null float64
GEO.display-label                            3142 non-null object
POPESTIMATE2016                              3142 non-null float64
AVG_NETMIG                                   3142 non-null float64
POP_CHANGE               

### Narrowing county list
<p>We should probably narrow this list down since it's pretty long. I people are leaving a county, it probably ios a bad place to live. So let's knock off all negative population growth counties.</p>

In [7]:
df = df[(df['AVG_NETMIG']>0)]

In [8]:
df = df[(df['POP_CHANGE']>0)]

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1078 entries, 0 to 3141
Data columns (total 25 columns):
Unnamed: 0                                   1078 non-null float64
SUMLEV                                       1078 non-null float64
REGION                                       1078 non-null float64
DIVISION                                     1078 non-null float64
STATE                                        1078 non-null float64
COUNTY                                       1078 non-null float64
STNAME                                       1078 non-null object
CTY                                          1078 non-null object
Unemployment_rate_2016                       1078 non-null float64
Med_HH_Income_Percent_of_State_Total_2015    1078 non-null float64
GEO.display-label                            1078 non-null object
POPESTIMATE2016                              1078 non-null float64
AVG_NETMIG                                   1078 non-null float64
POP_CHANGE               

<p>So after making sure we had positive net migration and a positive population change, we see we've drilled the list down to 1,078 counties.</p>

In [11]:
df.sort_values('MEDIAN_MORTAGE_PAYMENTS' , ascending=True)

Unnamed: 0  SUMLEV  REGION  DIVISION  STATE  COUNTY          STNAME  \
2673      2717.0    50.0     3.0       7.0   48.0   301.0           Texas   
2512      2555.0    50.0     3.0       6.0   47.0   169.0       Tennessee   
698        713.0    50.0     2.0       3.0   18.0     3.0         Indiana   
1722      1750.0    50.0     2.0       4.0   31.0   137.0        Nebraska   
3134      3185.0    50.0     4.0       8.0   56.0    31.0         Wyoming   
328        338.0    50.0     3.0       5.0   12.0    17.0         Florida   
2665      2709.0    50.0     3.0       7.0   48.0   285.0           Texas   
346        356.0    50.0     3.0       5.0   12.0    55.0         Florida   
1803      1835.0    50.0     4.0       8.0   35.0    15.0      New Mexico   
126        130.0    50.0     3.0       7.0    5.0    31.0        Arkansas   
1569      1595.0    50.0     2.0       4.0   29.0   173.0        Missouri   
2488      2531.0    50.0     3.0       6.0   47.0   121.0       Tennessee   
2472      2515.0    50.0     3.0       6.0   47.0    89.0       Tennessee   
694        708.0    50.0     2.0       3.0   17.0   199.0        Illinois   
339        349.0    50.0     3.0       5.0   12.0    41.0         Florida   
1022      1040.0    50.0     3.0       6.0   21.0    59.0        Kentucky   
2636      2680.0    50.0     3.0       7.0   48.0   227.0           Texas   
136        140.0    50.0     3.0       7.0    5.0    51.0        Arkansas   
272        278.0    50.0     4.0       8.0    8.0    55.0        Colorado   
1003      1021.0    50.0     3.0       6.0   21.0    21.0        Kentucky   
1118      1137.0    50.0     3.0       7.0   22.0    11.0       Louisiana   
2431      2474.0    50.0     3.0       6.0   47.0     7.0       Tennessee   
1509      1535.0    50.0     2.0       4.0   29.0    53.0        Missouri   
2435      2478.0    50.0     3.0       6.0   47.0    15.0       Tennessee   
2141      2178.0    50.0     3.0       7.0   40.0    21.0        Oklahoma   
2002      2037.0    50.0     2.0       4.0   38.0    25.0    North Dakota   
2193      2230.0    50.0     3.0       7.0   40.0   125.0        Oklahoma   
2192      2229.0    50.0     3.0       7.0   40.0   123.0        Oklahoma   
2670      2714.0    50.0     3.0       7.0   48.0   295.0           Texas   
2439      2482.0    50.0     3.0       6.0   47.0    23.0       Tennessee   
...          ...     ...     ...       ...    ...     ...             ...   
1644      1671.0    50.0     4.0       8.0   30.0    93.0         Montana   
1577      1603.0    50.0     2.0       4.0   29.0   187.0        Missouri   
1486      1512.0    50.0     2.0       4.0   29.0     7.0        Missouri   
2504      2547.0    50.0     3.0       6.0   47.0   153.0       Tennessee   
2177      2214.0    50.0     3.0       7.0   40.0    93.0        Oklahoma   
1911      1945.0    50.0     3.0       5.0   37.0    43.0  North Carolina   
2154      2191.0    50.0     3.0       7.0   40.0    47.0        Oklahoma   
1751      1780.0    50.0     4.0       8.0   32.0     9.0          Nevada   
2738      2782.0    50.0     3.0       7.0   48.0   431.0           Texas   
2535      2579.0    50.0     3.0       7.0   48.0    25.0           Texas   
1079      1097.0    50.0     3.0       6.0   21.0   173.0        Kentucky   
1099      1117.0    50.0     3.0       6.0   21.0   213.0        Kentucky   
2173      2210.0    50.0     3.0       7.0   40.0    85.0        Oklahoma   
2020      2055.0    50.0     2.0       4.0   38.0    61.0    North Dakota   
732        747.0    50.0     2.0       3.0   18.0    71.0         Indiana   
1010      1028.0    50.0     3.0       6.0   21.0    35.0        Kentucky   
2394      2436.0    50.0     2.0       4.0   46.0    67.0    South Dakota   
2356      2397.0    50.0     3.0       5.0   45.0    81.0  South Carolina   
2171      2208.0    50.0     3.0       7.0   40.0    81.0        Oklahoma   
2180      2217.0    50.0     3.0       7.0   40.0    99.0        Oklahoma   
13

### Z-Score analysis
<p>Now we will assign Z-scores to all of the counties to be able to cpmpare several measures.</p>
<p>We want counties with a growing population of families, not bad poverty, etc.</p>

In [ ]:
df['zpopchange']